### PR TITLE
New data source: azurerm_storage_account_blob_container_sas

### LINKED ISSUE #3068: Feature Request: Add support for Azure Container SAS tokens
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently there is no native support for Azure Service SAS tokens.

As already mentioned #59 there are different types of SAS tokens.

To allow a more fine-grained access control it would be nice to have native support inside terraform without the need to use local-exec. 

As the configuration options (especially the permission semantic) for Service SAS tokens depends on the service type i would suggest to create different SAS data sources for each of the following types: `blob`, `container`, `share`, `file`, `queue`, `table`

### New or Affected Resource(s)

* azurerm_storage_container_sas
* azurerm_storage_\<type\>_sas

### Potential Terraform Configuration

```hcl
data "azurerm_storage_container_sas" "foo" {
  connection_string = "DefaultEndpointsProtocol=https;..."
  https_only = true
  ip = "127.0.0.1" // optional

  container_name = "bar"

   // Response header settings
  cache-control = ""
  content-disposition = ""
  content-encoding =  ""
  content-language = ""
  content-type = ""

  start  = "2019-03-18" // optional
  expiry = "2020-01-01"

  permissions {
    read    = true
    write   = true
    delete  = true
    list    = true
    add     = true
    create  = true
  }

  // access policy as alternative for permissions and  validity period
  policy-name = ""
}

```

### References
* https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas
* https://docs.microsoft.com/en-us/cli/azure/storage/container?view=azure-cli-latest#az-storage-container-generate-sas
*  https://github.com/terraform-providers/terraform-provider-azurerm/issues/59#issuecomment-374761772

### PR DESCRIPTION
**Pull request by r0bnet**
Implements what's requested here: https://github.com/terraform-providers/terraform-provider-azurerm/issues/3068

Fixes #3068

### CODE REVIEW

File: azurerm/data_source_storage_account_blob_container_sas.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+			"content_language": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"content_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply azurerm/data_source_storage_account_blob_container_sas.go]: fixed everything :)

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2019-09-04 at 10 20 33

**Comment by ibayer** [comment]
Thanks for this great MR, fine grand SAS is really important for me.

@tombuildsstuff 
I couldn't find any fixed release cycle for this provider. When can we expect this MR to me included?
Even an tentative estimate would help me to decide if I need to use a workaround or wait for the next release.

**Comment by r0bnet** [comment]
Hey @ibayer 
happy to hear that this helps you. I heard rumors that they're aiming for releasing 1.34.0 somewhere next week. But no guarantee on this.

Here you can see the status / progress for next releases: https://github.com/terraform-providers/terraform-provider-azurerm/milestones

**Pull request merged**